### PR TITLE
Add parameter mapping rules for macros

### PR DESCRIPTION
**Pull request by gatecat**
This implements the second option discussed in https://github.com/SymbiFlow/fpga-interchange-schema/issues/41#issuecomment-824880077, that issue has some more background on the requirements here in general.

Note that this PR is on top of #40 to avoid an ordinal conflict that would otherwise occur.

### CODE REVIEW

**Review by clavin-xlnx — APPROVED** [review]
Looks good to me.

**Comment by gatecat** [comment]
Friendly bump, I don't _think_ any part of this or #40 that it is on top of is controversial?

**Review by acomodi — APPROVED** [review]
LGTM

File: interchange/DeviceResources.capnp [hunk 1]
```diff
@@ -234,14 +234,62 @@ struct Device {
...
+  }
+
   ######################################
   # Macro expansion exception map for
   # primitives that don't expand to a
-  # macro of the same name.
+  # macro of the same name. This is also
+  # used for conditional matches on
+  # parameter values and parameter
+  # transforms from primitiver to
```
[the remark this answers is not in the document]
Comment (gatecat) [reply interchange/DeviceResources.capnp]: Thanks, fixed

**Pull request merged**